src/module_wrap.cc

```diff
@@ -550,6 +550,62 @@ void ModuleWrap::Resolve(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(result.FromJust().ToObject(env));
 }
 
+static MaybeLocal<Promise> ImportModuleDynamically(
+    Local<Context> context,
+    Local<v8::ScriptOrModule> referrer,
+    Local<String> specifier) {
+  Isolate* iso = context->GetIsolate();
+  Environment* env = Environment::GetCurrent(context);
+  v8::EscapableHandleScope handle_scope(iso);
+
+  if (env->context() != context) {
+    auto maybe_resolver = Promise::Resolver::New(context);
+    Local<Promise::Resolver> resolver;
+    if (maybe_resolver.ToLocal(&resolver)) {
+      // TODO(jkrems): Turn into proper error object w/ code
+      Local<Value> error = v8::Exception::Error(
+        OneByteString(iso, "invalid_context"));
```

Comment:
Are there plans to support dynamic `import()` in VM contexts? If so, I'd rather the error be something to the sense of "Dynamic import() callback not registered for current context" than "invalid\_context".

Reply:
It... depends. Especially in [a future where Realms are a thing](https://github.com/tc39/proposal-realms), we'd want some sort of support for having per-context module maps. So this error is meant to be temporary, pending a decision around how and where to handle module imports in contexts other than the main one.

So depending on where that discussion goes, the actual error wording (and code) might be different. So I would rather defer introducing an opinion here for now. :)

Reply:
Thanks for the link to the proposal. I see reasons for your ambivalence about being too opinionated. However, I'd still prefer something a bit more user-friendly than "invalid\_context", maybe just "TypeError: import() called in VM context"? We can worry about `err.code` later.